src/Symfony/Component/HttpClient/Chunk/ErrorChunk.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient\Chunk;
+
+use Symfony\Component\HttpClient\Exception\TransportException;
+use Symfony\Contracts\HttpClient\ChunkInterface;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ *
+ * @internal
+ */
+class ErrorChunk implements ChunkInterface
+{
+    protected $throw;
+
+    private $offset;
+    private $error;
+
+    /**
+     * @param bool &$throw Allows monitoring when the $error has been thrown or not
+     */
+    public function __construct(bool &$throw, int $offset, \Throwable $error = null)
+    {
+        $throw = true;
+        $this->throw = &$throw;
+        $this->offset = $offset;
+        $this->error = null !== $error ? [$error->getMessage(), 0, $error] : ['Reading from the response stream reached the inactivity timeout.'];
```

Comment:
alternatively `$this->error = $error` (the property already holds it), then built the transaction exception lazy, i.e. 

```
throw $this->getTransactionException();
```

Reply:
we don't want to instantiate the exception in the constructor, but when it is thrown. Otherwise, it suffers 2 issues:

- the exception might not show the right stack trace when shown
- the ErrorChunk becomes part of a circular object graph due to the stack trace (which includes references to any caller object), and that would create issues. See https://github.com/symfony/symfony/pull/30413#discussion_r261713719

Storing a reference to `$error` would suffer from the second point.

Reply:
what @stof said
array now replaced by properties.